### PR TITLE
Add `PreferentialGPSampler` document

### PR DESCRIPTION
**Pull request by not522**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## What does this implement/fix? Explain your changes.

Add document of `PreferentialGPSampler`.

image

### CODE REVIEW

**Comment by not522** [comment]
@contramundum53 Could you review this PR?

File: optuna_dashboard/preferential/samplers/gp.py [hunk 1]
```diff
@@ -280,6 +280,25 @@ def sample_gp(self, x: Tensor, preferences: Tensor) -> _SampledGP:
...
 
 class PreferentialGPSampler(optuna.samplers.BaseSampler):
+    """Sampler for preferential optimization using Gaussian process.
+
+    The sampling algorithm is based on `Takeno et al., 2023 <https://arxiv.org/abs/2302.01513>`_.
+    This sampler uses BoTorch to optimize acquisition function.
+
+    Args:
+        kernel:
+            Kernel that computes the covariance on the Gaussian process.
```
Comment (contramundum53) [inline optuna_dashboard/preferential/samplers/gp.py:290]:
```suggestion
            Kernel that computes the covariance on the Gaussian process. Defaults to Matern 3/2 Kernel + ARD.
```

File: optuna_dashboard/preferential/samplers/gp.py [hunk 2]
```diff
@@ -280,6 +280,25 @@ def sample_gp(self, x: Tensor, preferences: Tensor) -> _SampledGP:
...
+        noise_prior:
+            Prior of the observation noise.
```
Comment (contramundum53) [inline optuna_dashboard/preferential/samplers/gp.py:292]:
```suggestion
            Prior of the observation noise. Defaults to gamma prior.
```

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Comment by contramundum53** [comment]
Oops, please fix the linter.

**Pull request merged**